webapp/src/snippetBuilder.tsx

```diff
@@ -207,99 +272,113 @@ export class SnippetBuilder extends data.Component<SnippetBuilderProps, SnippetB
         this.hide();
     }
 
+    getCurrentPage() {
+        const { history } = this.state;
+
+        return history[history.length - 1];
+    }
+
+    getCurrentQuestion() {
+        const { config } = this.state;
+
+        return config.questions[this.getCurrentPage()];
+    }
+
+    getNextQuestion() {
+        const { config } = this.state;
+        const currentQuestion = this.getCurrentQuestion();
+        if (currentQuestion.goto) {
+            return config.questions[currentQuestion.goto.question];
+        }
+        return null;
+    }
+
+    isLastQuestion() {
+        if (this.getCurrentQuestion().goto) {
+            return false;
+        }
+
+        return true;
+    }
+
     /**
      * Changes page by 1 if next question exists.
      * Looks for output and appends the next questions output if it exists and
      * is not already attached to the current output.
      */
     nextPage() {
-        const { config } = this.state;
         const { history, tsOutput } = this.state;
```

Comment:
Add tick

Reply:
Yeah definitely. I'm going to hopefully time each question as well.

Reply:
It is interresting to know if people go back and forth, or always forward. Put ticks on meaningful user interactions. Also add metadata to track completion. Can be done in future PR.

Reply:
Alright sounds good. I'm going to get it in this PR so that when we start testing I'll be able to look at the analytics.